app/pages/amigo-requests/amigo-requests.ts

```diff
@@ -49,11 +53,30 @@ export class AmigoRequestsPage implements OnInit {
      * @returns {any}.
      */
     public loadRequests(): any {
-        return this.amigoShareRequestService.list().then(
-            amigoShareRequests => {
+
+        const loadingStatus: LoadingStatus = this.loadingStatusService.start(this.nav);
+
+        const listPromise = this.amigoShareRequestService.list().then(
+            (amigoShareRequests) => {
                 this.amigoShareRequests = amigoShareRequests;
             }
         );
+
+        /* Wait on request to resolve, and loading mask to display, then stop loading status. */
+        return Promise.all(
+            [
+                listPromise,
+                loadingStatus.displayPromise
+            ]
+        ).then(
+            () => {
+                loadingStatus.loading.dismiss();
+            }
+        ).catch(
+            () => {
+                loadingStatus.loading.dismiss();
+            }
+        );
```

Comment:
This is the pattern I came up with to avoid timing issues with Ionics loading mask.

Showing the loading UI returns a promise that resolves when it is actually displayed. We wait on both this display promise and the actual server call promise, and when both are done it is safe to dismiss the loading UI.